Mono.Debugging.Soft/SoftDebuggerSession.cs

```diff
@@ -737,19 +739,84 @@ protected override ProcessInfo[] OnGetProcesses ()
 		internal PortablePdbData GetPdbData (AssemblyMirror asm)
 		{
 			string assemblyFileName;
+
 			if (!assemblyPathMap.TryGetValue (asm.GetName ().FullName, out assemblyFileName))
 				assemblyFileName = asm.Location;
 			var pdbFileName = Path.ChangeExtension (assemblyFileName, ".pdb");
-			if (!PortablePdbData.IsPortablePdb (pdbFileName))
-				return null;
-			return new PortablePdbData (pdbFileName);
+			if (PortablePdbData.IsPortablePdb (pdbFileName))
+				return new PortablePdbData (pdbFileName);
+			// Attempt to fetch pdb from the debuggee over the wire
+			var pdbBlob = asm.GetPdbBlob ();
+			return pdbBlob != null ? new PortablePdbData (pdbBlob) : null;
 		}
 
 		internal PortablePdbData GetPdbData (MethodMirror method)
 		{
 			return GetPdbData (method.DeclaringType.Assembly);
 		}
 
+		Dictionary<AssemblyMirror, SourceLinkMap []> SourceLinkCache = new Dictionary<AssemblyMirror, SourceLinkMap []> ();
```

Comment:
Should we actually use a ConditionalWeakTable here? We want the SourceLinkMap to last as much as the AssemblyMirror, so if the AssemblyMirror does end up collected, we want the SourceLinkMap to be collected too?

Reply:
I don't think it matters, AssemblyMirror and DebuggerSession have the same lifespan.